### PR TITLE
make hmdb train/val list generation more robust

### PR DESCRIPTION
**Pull request by dreamerlin**

### CODE REVIEW

**Comment by kennymckormick** [comment]
Perhaps I didn't get it. Why the old codes are less robust and how this PR is related to Issue #808. Can you add some explanation to help me understand it?

**Comment by dreamerlin** [comment]
@kennymckormick If 'data/hmdb51/annotations/classInd.txt' is generated once and `data/hmdb51/annotations/trainlist{:02d}.txt` / `data/hmdb51/annotations/testlist{:02d}.txt` happen to be lost (program crash, file transmisson, etc).  These two lines will not be go through.

https://github.com/open-mmlab/mmaction2/blob/8972e537a77770c4488b33aa595ae45054239159/tools/data/parse_file_list.py#L472-L473

which may cause the failure of generating `data/hmdb51/annotations/trainlist{:02d}.txt` / `data/hmdb51/annotations/testlist{:02d}.txt`

File: tools/data/parse_file_list.py [hunk 1]
```diff
@@ -430,15 +430,23 @@ def generate_class_index_file():
...
+                    cur_line = ' '.join([str(class_id + 1), class_name])
+                    content.append(cur_line)
+                content = '\n'.join(content)
+                f.write(content)
+        else:
+            print(f'{class_index_file} has been generated before.')
+            class_dict = {
+                class_name: class_id + 1
+                for class_id, class_name in enumerate(class_list)
+            }
```
Comment (congee524) [inline tools/data/parse_file_list.py:449]: we can define class_dict first and once
```python
class_dict = {
    class_name: class_id + 1
    for class_id, class_name in enumerate(class_list)
}
if not osp.exists(class_index_file):
    ...
else:
    ...
```

**Comment by congee524** [comment]
Maybe it's caused by other problems. We can keep the original issue #808 first

**Review by congee524 — APPROVED** [review]

**Comment by kennymckormick** [comment]
> @kennymckormick If 'data/hmdb51/annotations/classInd.txt' is generated once and `data/hmdb51/annotations/trainlist{:02d}.txt` / `data/hmdb51/annotations/testlist{:02d}.txt` happen to be lost (program crash, file transmisson, etc). These two lines will not be go through.
> 
> https://github.com/open-mmlab/mmaction2/blob/8972e537a77770c4488b33aa595ae45054239159/tools/data/parse_file_list.py#L472-L473
> 
> which may cause the failure of generating `data/hmdb51/annotations/trainlist{:02d}.txt` / `data/hmdb51/annotations/testlist{:02d}.txt`

got it

**Review by kennymckormick — APPROVED** [review]

**Pull request merged**